### PR TITLE
#126 Add AndroidOrientationService

### LINKED ISSUE #126: Implement Orientation service for Android

### PR DESCRIPTION
**Pull request by jperedadnr**
Fixes #126

### CODE REVIEW

File: modules/orientation/src/main/java/com/gluonhq/attach/orientation/impl/AndroidOrientationService.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    private static Optional<Orientation> convertOrientation(String text) {
+        orientationText = text;
+        switch (orientationText) {
+            case "Portrait":
+                return Optional.of(Orientation.VERTICAL);
+            case "Landscape":
+                return Optional.of(Orientation.HORIZONTAL);
+            case "Unknown":
```
Comment (tiainen) [inline modules/orientation/src/main/java/com/gluonhq/attach/orientation/impl/AndroidOrientationService.java:76]: "Unknown" is already handled by the default case.

File: modules/orientation/src/main/native/android/c/Orientation.c [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+// Graal handles
+static jclass jGraalOrientationClass;
+static jmethodID jGraalNotifyOrientationMethod;
+
+static void initializeGraalHandles(JNIEnv* env) {
+    jGraalOrientationClass = (*env)->NewGlobalRef(env, (*env)->FindClass(env, "com/gluonhq/attach/orientation/impl/AndroidOrientationService"));
+    jGraalNotifyOrientationMethod = (*env)->GetStaticMethodID(env, jGraalOrientationClass, "notifyOrientation", "(Ljava/lang/String;)V");
+}
+
+static void initializeOrientationDalvikHandles() {
```
Comment (tiainen) [inline modules/orientation/src/main/native/android/c/Orientation.c]: From a quick look, it almost looked to me like this method is no longer required. But it does create an instance of the dalvik orientiation class, to set the lifecycle handler and intent receiver. Perhaps the method should be renamed as it doesn't initialize any class/method/field handles. It actually initializes the dalvik orientation class.

File: modules/orientation/src/main/java/com/gluonhq/attach/orientation/impl/AndroidOrientationService.java (hunk 1 above)
Comment (jperedadnr) [reply modules/orientation/src/main/java/com/gluonhq/attach/orientation/impl/AndroidOrientationService.java:76]: Indeed, but I've added it there, so we explicitly match the strings we send from the dalvik side.

File: modules/orientation/src/main/native/android/c/Orientation.c (hunk 2 above)
Comment (jperedadnr) [reply modules/orientation/src/main/native/android/c/Orientation.c]: The method is required to create the `DalvikOrientationService` instance, by getting its class/method handles only locally. 
I don't see a need to rename it. We use the same convention for all the services (whether we create global handles or not). Imo, it is better to keep it

Comment (tiainen) [reply modules/orientation/src/main/native/android/c/Orientation.c]: Well, the reason I brought it up, was that I always thought these methods were used to cache some dalvik handles for later use. But it seems it also does some initialization. Which means this method does more then its name is suggesting. Which in turn made me have to look into the method details when I saw it didn't save any of the handles it looked up. Ideally, I would separate the things that load handles for later use from things that do initialization.

Comment (jperedadnr) [reply modules/orientation/src/main/native/android/c/Orientation.c]: So you want to split `initialize<ServiceName>DalvikHandles` into something like `initialize<ServiceName>DalvikService` and `initialize<ServiceName>DalvikHandles`?
That's for a whole new refactoring...

Comment (tiainen) [reply modules/orientation/src/main/native/android/c/Orientation.c]: That was the idea yes, at least it is currently hard to see what the method is used for actually.

**Review by tiainen — APPROVED** [review]

File: modules/orientation/src/main/native/android/c/Orientation.c (hunk 2 above)
Comment (jperedadnr) [reply modules/orientation/src/main/native/android/c/Orientation.c]: I'll file an issue for a follow-up PR.

**Pull request merged**